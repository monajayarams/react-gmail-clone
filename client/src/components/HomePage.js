import React, { useState } from 'react'
import { Toaster } from 'react-hot-toast'
import Header from './Header'
import { Sidebar } from './Sidebar'
import { MailContainer } from './Mail'
import { ModalContainer } from './Modal'
import { HomePageContainer, HomePageContent } from './StyledComponents'


export const HomePage = ({ handleLogout }) => {
  const [modalOpen, setModalOpen] = useState(false)
  const [selectedFolder, setSelectedFolder] = useState('inbox')

  const sideIconClickHanlder = (name) => {
    setSelectedFolder(name)
  }
  const showHideModal = () => {
    setModalOpen(!modalOpen)
  }

  return (
    <HomePageContainer>
      <Header handleLogout={handleLogout} />
      <HomePageContent>
        <Toaster
          toastOptions={{
            duration: 5000,
            success: {
              iconTheme: {
                primary: 'gray',
                secondary: 'white',
              },
            },
            error: {
              iconTheme: {
                primary: 'red',
                secondary: 'white',
              },
            },
          }}
        />
        <Sidebar
          selectedFolder={selectedFolder}
          showHideModal={showHideModal}
          sideIconClickHanlder={sideIconClickHanlder}
        />
        <MailContainer selectedFolder={selectedFolder} />
        {modalOpen && <ModalContainer showHideModal={showHideModal}/>}
      </HomePageContent>
    </HomePageContainer>
  );
}
