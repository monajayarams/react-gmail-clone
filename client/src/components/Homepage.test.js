import React from 'react';
import { act, fireEvent, render } from '@testing-library/react';
import { HomePage } from './HomePage';

describe('verify Homepage Snapshot - Mail List', () => {
  it('loads homepage with mail list', () => {
    const Homepage = render(<HomePage />)
    expect(Homepage).toMatchSnapshot()
  })
})

describe('verify HomePage Snapshot - Compose Mail', () => {
  it('loads Compose mail modal', async() => {
    const { getByText } = render(<HomePage />)
    const composeBtn = getByText('Compose')
    act(() => {
      fireEvent.click(composeBtn)
    })
    expect(() => getByText('New Message')).not.toThrow()
  })
})