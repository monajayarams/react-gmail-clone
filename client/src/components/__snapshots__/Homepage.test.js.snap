// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`verify Homepage Snapshot - Mail List loads homepage with mail list 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-eCYdqJ bcsQqv"
      >
        <div
          class="sc-bczRLJ iaZVyv"
        >
          <div
            class="sc-gsnTZi hcFwrN"
          >
            <img
              alt="gmailLogo"
              src="https://ssl.gstatic.com/ui/v1/icons/mail/rfr/logo_gmail_lockup_default_2x_r2.png"
            />
          </div>
          <div
            class="sc-dkzDqf hoZXih"
          >
            <div
              class="sc-hKMtZM klGwhQ"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiAvatar-fallback"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-jSMfEi jlLnho"
        >
          <div
            style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
          />
          <div
            class="sc-dmRaPn vNkzi"
          >
            <div
              class="sc-kgflAQ eKUBVk"
            >
              <div
                class="sc-bBrHrO dimRlM"
              >
                Compose
              </div>
              <div
                class="sc-fLlhyt hrrwmv active"
              >
                <div
                  class="sc-jOrMOR cdqhMV"
                />
                <div>
                  Inbox
                </div>
              </div>
              <div
                class="sc-fLlhyt hrrwmv"
              >
                <div
                  class="sc-bZnhIo jcOsAa"
                />
                <div>
                  Starred
                </div>
              </div>
              <div
                class="sc-fLlhyt hrrwmv"
              >
                <div
                  class="sc-dPyBCJ mIOCD"
                />
                <div>
                  Sent
                </div>
              </div>
              <div
                class="sc-fLlhyt hrrwmv"
              >
                <div
                  class="sc-iTONeN ciFNvj"
                />
                <div>
                  Drafts
                </div>
              </div>
              <div
                class="sc-fLlhyt hrrwmv"
              >
                <div
                  class="sc-cTQhss iHhiWz"
                />
                <div>
                  Important
                </div>
              </div>
              <div
                class="sc-fLlhyt hrrwmv"
              >
                <div
                  class="sc-iAvgwm gEGVRo"
                />
                <div>
                  Spam
                </div>
              </div>
              <div
                class="sc-fLlhyt hrrwmv"
              >
                <div
                  class="sc-efBctP kIzkyi"
                />
                <div>
                  Trash
                </div>
              </div>
            </div>
          </div>
          <div
            class="sc-iBkjds fFQuhq"
          >
            <div
              class="sc-ftvSup bFyVIz"
            >
              <div
                class="sc-gKXOVf cekoTV"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="sc-papXJ fmRhqr"
            >
              <div>
                <div
                  class="mailCards"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-eCYdqJ bcsQqv"
    >
      <div
        class="sc-bczRLJ iaZVyv"
      >
        <div
          class="sc-gsnTZi hcFwrN"
        >
          <img
            alt="gmailLogo"
            src="https://ssl.gstatic.com/ui/v1/icons/mail/rfr/logo_gmail_lockup_default_2x_r2.png"
          />
        </div>
        <div
          class="sc-dkzDqf hoZXih"
        >
          <div
            class="sc-hKMtZM klGwhQ"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiAvatar-fallback"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-jSMfEi jlLnho"
      >
        <div
          style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
        />
        <div
          class="sc-dmRaPn vNkzi"
        >
          <div
            class="sc-kgflAQ eKUBVk"
          >
            <div
              class="sc-bBrHrO dimRlM"
            >
              Compose
            </div>
            <div
              class="sc-fLlhyt hrrwmv active"
            >
              <div
                class="sc-jOrMOR cdqhMV"
              />
              <div>
                Inbox
              </div>
            </div>
            <div
              class="sc-fLlhyt hrrwmv"
            >
              <div
                class="sc-bZnhIo jcOsAa"
              />
              <div>
                Starred
              </div>
            </div>
            <div
              class="sc-fLlhyt hrrwmv"
            >
              <div
                class="sc-dPyBCJ mIOCD"
              />
              <div>
                Sent
              </div>
            </div>
            <div
              class="sc-fLlhyt hrrwmv"
            >
              <div
                class="sc-iTONeN ciFNvj"
              />
              <div>
                Drafts
              </div>
            </div>
            <div
              class="sc-fLlhyt hrrwmv"
            >
              <div
                class="sc-cTQhss iHhiWz"
              />
              <div>
                Important
              </div>
            </div>
            <div
              class="sc-fLlhyt hrrwmv"
            >
              <div
                class="sc-iAvgwm gEGVRo"
              />
              <div>
                Spam
              </div>
            </div>
            <div
              class="sc-fLlhyt hrrwmv"
            >
              <div
                class="sc-efBctP kIzkyi"
              />
              <div>
                Trash
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-iBkjds fFQuhq"
        >
          <div
            class="sc-ftvSup bFyVIz"
          >
            <div
              class="sc-gKXOVf cekoTV"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
                />
              </svg>
            </div>
          </div>
          <div
            class="sc-papXJ fmRhqr"
          >
            <div>
              <div
                class="mailCards"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
